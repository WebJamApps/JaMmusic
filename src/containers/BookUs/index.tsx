import { Button } from '@mui/material';
import { Inquiry } from '../NewHomepage/Inquiry';

const songlistUrl = 'https://docs.google.com/spreadsheets/d/1uePCuHO6qxg2tHTqGist_O4iYXJjy9AgSmXvMgF9NPM/edit?usp=sharing';
export function BookUs() {
  return (
    <div className="bookus" style={{ margin: 'auto', padding: '6px', paddingRight: '12px' }}>
      <div style={{ maxWidth: '800px' }}>
        <h3 style={{ textAlign: 'center' }}>Book Us</h3>
        <h4 style={{ fontWeight: 'normal' }}>
          We are always looking for regional performance opportunites,
          and would consider traveling even further away from our home in Salem, VA for just the right situation.
          <Button
            style={{ marginLeft: '8px' }}
            size="small"
            variant="contained"
            onClick={() => window.open(songlistUrl)}
          >
            Current Songlist
          </Button>

        </h4>
        <p style={{ textAlign: 'center' }}>
          Please fill out this form and we will get back with you soon.
        </p>
        <Inquiry country="United States" hideTitle />
      </div>
    </div>
  );
}
