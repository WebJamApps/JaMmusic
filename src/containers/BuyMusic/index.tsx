import { Component } from 'react';
import Jsb from './BuyLinks/joshShermanBand';
import Jss from './BuyLinks/joshShermanSolo';
import Jsyt from './MediaLinks/joshShermanYoutube';
import Jssp from './MediaLinks/joshShermanSpotify';
import Jsam from './MediaLinks/joshShermanApple';
import commonUtils from '../../lib/utils';

export default class BuyMusic extends Component {
  commonUtils: { setTitleAndScroll: (pageTitle: string, width: number) => void };

  constructor(props: Record<string, unknown>) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount(): void { this.commonUtils.setTitleAndScroll('Buy Music', window.screen.width); }

  render(): JSX.Element {
    return (
      <div
        className="page-content"
        style={{
          paddingRight: '15px', maxWidth: '800px', margin: 'auto', borderRight: 'none',
        }}
      >
        <h2 style={{
          textAlign: 'center', margin: '20px', fontWeight: 'bold', marginBottom: '10px',
        }}
        >
          Buy from CD Baby
        </h2>
        <div className="row top-row-style">
          <Jsb />
          <Jss />
        </div>
        <Jssp />
        <Jsyt />
        <Jsam />
      </div>
    );
  }
}
