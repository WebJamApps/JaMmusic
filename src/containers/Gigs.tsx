import React, { useContext, useState } from 'react';
import { GigsContext } from '../providers/Gigs.provider';

export const Gigs = (): JSX.Element => {
  const { gigs } = useContext(GigsContext);
  console.log(gigs);
  const [allGigs, setGigs] = useState(gigs);
  return (
    <div style={{ margin: 'auto' }}>
      <h3>Gigs</h3>
      {JSON.stringify(allGigs)}
    </div>
  );
};
