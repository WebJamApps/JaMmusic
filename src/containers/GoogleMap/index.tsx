
import { Component } from 'react';
import UMap from './UserMap';
import CMap from './CompanyMap';
import gMapUtils from './gMapUtils';

export interface Iloc { lat: number, lng: number }
export const equals = () => true;
class GoogleMap extends Component {
  gMap: google.maps.Map | null;

  companyMap: CMap | null;

  userMap: UMap | null;

  content: string;

  gMapUtils: typeof gMapUtils;

  constructor(props: Record<string, unknown>) {
    super(props);
    this.gMap = null;
    this.userMap = null;
    this.companyMap = null;
    this.content = '';
    this.gMapUtils = gMapUtils;
  }

  componentDidMount(): void {
    const mapDiv = document.getElementById('googleMap');
    if (mapDiv) this.gMap = new google.maps.Map(mapDiv, { zoom: 2, center: { lat: 40, lng: -100 } });
    this.userMap = new UMap();
    this.userMap = this.gMapUtils.limitUserLat(this.userMap);
    this.content = `<div><p><strong>User Name:</strong> ${this.userMap.name}</p>`
      + `<p><strong>Latitude:</strong> ${this.userMap.loc.lat}</p></div>`;
    this.addMarker(this.userMap);
    this.companyMap = new CMap();
    this.companyMap = this.gMapUtils.limitCompanyLat(this.companyMap);
    this.content = `<div><p><strong>Company Name:</strong> ${this.companyMap.name}</p>`
      + `<p><strong>Company Slogan:</strong> ${this.companyMap.catchPhrase}</p>`
      + `<p><strong>Latitude:</strong> ${this.companyMap.loc.lat}</p></div>`;
    this.addMarker(this.companyMap, {
      scaledSize: { width: 40, height: 40, equals },
      url: 'https://image.flaticon.com/icons/png/512/63/63838.png',
    });
  }

  addMarker(obj: { loc: Iloc }, icon?: google.maps.Icon | undefined): void {
    const infoWindow = new google.maps.InfoWindow({ content: this.content });
    if (this.gMap !== null) {
      const marker = new google.maps.Marker({
        map: this.gMap,
        position: {
          lat: obj.loc.lat, lng: obj.loc.lng,
        },
        icon,
      });
      marker.addListener('click', () => { // eslint-disable-next-line security/detect-non-literal-fs-filename
        /* istanbul ignore else */if (this.gMap !== null)infoWindow.open(this.gMap, marker);
      });
    }
  }

  render(): JSX.Element {
    return (
      <div className="page-content google">
        <div style={{ margin: 'auto', textAlign: 'center' }}>
          <h3 style={{ marginTop: '10px' }}>Google Map Here</h3>
          <div id="googleMap" style={{ height: '8.5in', maxWidth: '11in', margin: 'auto' }} />
        </div>
      </div>
    );
  }
}
export default GoogleMap;
