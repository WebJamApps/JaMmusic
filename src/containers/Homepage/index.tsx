import React, { Component } from 'react';
import ReactResizeDetector from 'react-resize-detector';
import WideAboutUs from './Widescreen/WideAbout';
import WideCurrentProjects from './Widescreen/WideCurrentProjects';
import NarrowAboutUs from './Narrowscreen/NarrowAbout';
import NarrowCurrentProjects from './Narrowscreen/NarrowCurrentProjects';
import FacebookFeed from './Narrowscreen/NarrowFacebookFeed';
import Inquiry from '../../components/inquiry';
import commonUtils from '../../lib/commonUtils';

interface HomepageState { width: number }

export default class Home extends Component<{}, HomepageState> {
  parentRef: any;

  commonUtils: any;

  constructor(props: any) {
    super(props);
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 320 };
    this.commonUtils = commonUtils;
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('', window.screen.width); }

  onResize(width: number) {
    this.setState({ width });
  }

  render() {
    const { width } = this.state;
    return (
      <div>
        {width >= 1004
          ? (
            <div className="page-content">
              <div className="anchor"> </div>
              <WideAboutUs />
              <hr />
              <WideCurrentProjects />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )
          : (
            <div className="page-content">
              <NarrowAboutUs />
              <hr />
              <NarrowCurrentProjects />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
              <FacebookFeed />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
              <Inquiry />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />
      </div>
    );
  }
}
