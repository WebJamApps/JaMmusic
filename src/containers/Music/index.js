import React, { Component, useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PicSlider from '../../components/pic-slider';
import DefaultTable from '../../components/tour-table';
import JoshBio from './joshBio';
import MariaBio from './mariaBio';
import EmersonBio from './emersonBio';
import BrianBio from './brianBio';
import Wjband from './wjBand';
import Intro from './intro';
 import mapStoreToProps from '../../redux/mapStoreToProps';
 // import MusicDashboard from '../MusicDashboard/index'; 

export class Music extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() { document.title = 'Music | Web Jam LLC'; }

  musicians() { // eslint-disable-line class-methods-use-this
    return (
      <div className="elevation3" style={{ maxWidth: '1000px', margin: 'auto' }}>
        <section>
          <JoshBio />
          <hr />
          <MariaBio />
          <hr />
          <Wjband />
          <hr />
          <EmersonBio />
          <hr />
          <BrianBio />
        </section>
        <p>{' '}</p>
        <p>{' '}</p>
      </div>
    );
  }

  render() {
    const { images } = this.props;
    return (
      <div className="page-content">
        <div style={{ paddingTop: '1px', paddingBottom: 0, marginBottom: 0 }}>
          <div id="musicSlide1">
            {images.length > 0 ? (<PicSlider data={images} />) : null}
          </div>
          <Intro />
        </div>
        <div
          className="search-table-outer"
          style={{
            position: 'relative', overflowX: 'auto', maxWidth: '96%', margin: 'auto',
          }}
        >
        {/* <DefaultTable /> */}
        </div>
        <div style={{ height: '10px' }}>
          <p>{' '}</p>
        </div>
        {this.musicians()}
        {/* <MusicDashboard /> */}
      </div>
    );
  }
}

Music.defaultProps = { images: [] };
Music.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string,
    title: PropTypes.string,
  })),
};

export default connect(mapStoreToProps)(Music);
