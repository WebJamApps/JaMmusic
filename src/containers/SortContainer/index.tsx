import { gql, useQuery, ApolloError } from '@apollo/client';

function makeRandomString(): string {
  const randomString = Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15);
  const withNoDigits = randomString.replace(/[0-9]/g, 'X');
  return withNoDigits;
}

export const TEST_GQL = gql`
  query Gqldocs {
    gqldocs {
      title
      author
    }
  }
`;

interface IsortSectionProps {
  error: ApolloError | undefined, data: Record<string, unknown>, dice: number[], sorted: number[], randomString: string,
  loading:boolean
}
export function SortSection({
  error, data, dice, sorted, randomString, loading,
}: IsortSectionProps) {
  if (loading) return <p>Loading...</p>;
  if (error) {
    return (
      <p>
        Error :
        {' '}
        {error.message}
      </p>
    );
  }
  return (
    <div id="sort-container" style={{ margin: 'auto', textAlign: 'center' }}>
      <h4>Sorting Example</h4>
      <p style={{ textAlign: 'left', marginLeft: '10px' }}>
        Given this random number array:
        {' '}
        <strong>{dice}</strong>
        <br />
        This is the sorted array:
        {' '}
        <strong>{sorted}</strong>
        <br />
        <br />
        Given this random string of letters:
        {' '}
        <strong>{randomString}</strong>
        <br />
        This is the sorted string:
        {' '}
        <strong>
          {randomString.split('').sort((a, b) => a.localeCompare(b, 'en', { sensitivity: 'base' })).join('')}
        </strong>
      </p>
      <p>{JSON.stringify(data)}</p>
    </div>
  );
}

export function SortContainer(): JSX.Element {
  const { loading, error, data } = useQuery(TEST_GQL);
  const dice = Array.from({ length: 6 }, () => Math.floor(Math.random() * 6));
  const sorted = [...dice].sort();
  const randomString = makeRandomString();
  return <SortSection loading={loading} error={error} data={data} dice={dice} sorted={sorted} randomString={randomString} />;
}

