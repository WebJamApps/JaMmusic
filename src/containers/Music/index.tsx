import { useEffect, useState, useContext } from 'react';
import { AuthContext, Iauth } from 'src/providers/Auth.provider';
import { PicSlider } from 'src/containers/Music/Pictures/PicSlider';
import commonUtils from 'src/lib/utils';
import { Gigs } from './Gigs';
import { Intro } from './intro';
import JoshBio from './joshBio';
import MariaBio from './mariaBio';
import { CreatePicDialog } from './Pictures/CreatePicDialog';
import { EditPicTable } from './Pictures/EditPicTable';

export function Musicians(): JSX.Element {
  return (
    <div className="elevation3" style={{ maxWidth: '1000px', margin: 'auto' }}>
      <section>
        <JoshBio />
        <hr />
        <MariaBio />
      </section>
      <p>{' '}</p>
      <p>{' '}</p>
    </div>
  );
}

interface IphotosSectionProps {
  showEditPicTable: boolean, setShowEditPicTable: (arg0: boolean) => void, isAdmin: boolean,
  setShowCreatePic: (arg0: boolean) => void, showCreatePic: boolean,
}
export function PhotosSection(props: IphotosSectionProps) {
  const {
    showEditPicTable, setShowEditPicTable, isAdmin, setShowCreatePic, showCreatePic,
  } = props;
  return (
    <div style={{ paddingTop: '1px', paddingBottom: 0, marginBottom: 0 }}>
      <div id="musicSlide1">
        {!showEditPicTable ? <PicSlider /> : <EditPicTable setShowTable={setShowEditPicTable} />}
      </div>
      {showEditPicTable ? null
        : <Intro isAdmin={isAdmin} setShowEditPic={setShowEditPicTable} setShowCreatePic={setShowCreatePic} />}
      <CreatePicDialog showDialog={showCreatePic} setShowDialog={setShowCreatePic} />
    </div>
  );
}

export function checkIsAdmin(auth: Iauth, setIsAdmin: (arg0: boolean) => void) {
  let isAdmin = false;
  if (auth && auth.isAuthenticated && process.env.userRoles) {
    const { user: { userType } } = auth;
    const rolesJSON = JSON.parse(process.env.userRoles);
    const { roles } = rolesJSON;
    if (userType && roles.includes(userType)) isAdmin = true;
  }
  setIsAdmin(isAdmin);
}

export function Music(): JSX.Element {
  const { auth } = useContext(AuthContext);
  const [isAdmin, setIsAdmin] = useState(false);
  const [showCreatePic, setShowCreatePic] = useState(false);
  const [showEditPicTable, setShowEditPicTable] = useState(false);
  useEffect(() => commonUtils.setTitleAndScroll('Music', window.screen.width), []);
  useEffect(() => {
    checkIsAdmin(auth, setIsAdmin);
  }, [auth]);
  return (
    <div className="page-content music">
      <PhotosSection
        showEditPicTable={showEditPicTable}
        setShowEditPicTable={setShowEditPicTable}
        isAdmin={isAdmin}
        setShowCreatePic={setShowCreatePic}
        showCreatePic={showCreatePic}
      />
      <div
        className="search-table-outer"
        style={{
          position: 'relative', overflowX: 'auto', maxWidth: '96%', margin: 'auto', zIndex: 0,
        }}
      >
        {showEditPicTable ? null : <Gigs isAdmin={isAdmin} />}
      </div>
      <div style={{ height: '10px' }}>
        <p>{' '}</p>
      </div>
      {showEditPicTable ? null : <Musicians />}
    </div>
  );
}
