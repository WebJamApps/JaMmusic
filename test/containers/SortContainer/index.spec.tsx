import { SortContainer, SortSection, TEST_GQL } from 'src/containers/SortContainer';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { ApolloError } from '@apollo/client';

describe('SortContainer', () => {
  it('renders loading...', () => {
    const mocks = [
      {
        request: {
          query: TEST_GQL,
          variables: {
            name: 'Buck',
          },
        },
        result: {
          data: {
            dog: { id: '1', name: 'Buck', breed: 'bulldog' },
          },
        },
      },
    ];
    const sc:any = renderer.create(<MockedProvider mocks={mocks} addTypename={false}><SortContainer /></MockedProvider>).toJSON();
    expect(sc.children[0]).toBe('Loading...');
  });
  it('renders SortSection when error', () => {
    const props = {
      error: new ApolloError({ graphQLErrors: ['failed' as any] }), data: {}, dice: [], sorted: [], randomString: '', loading: false,
    };
    const ss = renderer.create(<SortSection {...props} />).toJSON() as ReactTestRendererJSON;
    expect(ss.type).toBe('p');
  });
  it('renders SortSection as div', () => {
    const props = {
      error: undefined, data: {}, dice: [], sorted: [], randomString: '', loading: false,
    };
    const ss = renderer.create(<SortSection {...props} />).toJSON() as ReactTestRendererJSON;
    expect(ss.type).toBe('div');
  });
});
