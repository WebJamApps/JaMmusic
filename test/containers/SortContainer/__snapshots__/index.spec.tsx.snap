// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortContainer renders correctly 1`] = `
<div
  id="sort-container"
  style={
    {
      "margin": "auto",
      "textAlign": "center",
    }
  }
>
  <h4>
    Sorting Example
  </h4>
  <p
    style={
      {
        "marginLeft": "10px",
        "textAlign": "left",
      }
    }
  >
    Given this random number array:
     
    <strong>
      2
      2
      0
      3
      3
      3
    </strong>
    <br />
    This is the sorted array:
     
    <strong>
      0
      2
      2
      3
      3
      3
    </strong>
    <br />
    <br />
    Given this random string of letters:
     
    <strong>
      wcqlwXtbXarXpzhjXbatnX
    </strong>
    <br />
    This is the sorted string:
     
    <strong>
      aabbchjlnpqrttwwXXXXXz
    </strong>
  </p>
</div>
`;
